Sources/BraveNews/Composer/FeedDataSource.swift

```diff
@@ -108,10 +108,13 @@ public class FeedDataSource: ObservableObject {
 
   private let todayQueue = DispatchQueue(label: "com.brave.today")
   private let reloadQueue = DispatchQueue(label: "com.brave.today.reload")
+  private weak var historyAPI: BraveHistoryAPI?
 
   // MARK: - Initialization
 
-  public init() {
+  public init(historyAPI: BraveHistoryAPI? = nil) {
+    self.historyAPI = historyAPI
```

Comment:
Is there a reason to allowing this to be nil? If this can be `nil` then we should just have it assigned like we do `BraveAds`

Reply:
The reason was it is used inside previewes and we have no way to mock apis bit I used it as you asked like BraveAds Change in https://github.com/brave/brave-ios/pull/7087/commits/7f0d662a53231d133f4d69b9fee05c97c988603a